MonoGame.Framework/Graphics/GraphicsDevice.OpenGL.cs

```diff
@@ -521,6 +521,25 @@ internal void PlatformCreateRenderTarget(Texture renderTarget, int width, int he
                     renderTarget2D.glColorBuffer = renderTarget2D.glTexture;
                 renderTarget2D.glDepthBuffer = depth;
                 renderTarget2D.glStencilBuffer = stencil;
+
+                if (preferredDepthFormat != DepthFormat.None)
+                {
+                    renderTarget2D.depthTexture = new Texture2D(this, width, height, false, SurfaceFormat.U248, Texture2D.SurfaceType.Texture);
+                    GL.BindTexture(TextureTarget.Texture2D, renderTarget2D.depthTexture.glTexture);
+                    GraphicsExtensions.CheckGLError();
+
+                    GL.TexParameter(TextureTarget.Texture2D, TextureParameterName.TextureMinFilter, (int)TextureMinFilter.Nearest);
+                    GraphicsExtensions.CheckGLError();
+
+                    GL.TexParameter(TextureTarget.Texture2D, TextureParameterName.TextureMagFilter, (int)TextureMagFilter.Nearest);
+                    GraphicsExtensions.CheckGLError();
+
+                    GL.TexParameter(TextureTarget.Texture2D, TextureParameterName.TextureWrapS, (int)TextureWrapMode.ClampToEdge);
+                    GraphicsExtensions.CheckGLError();
+
+                    GL.TexParameter(TextureTarget.Texture2D, TextureParameterName.TextureWrapT, (int)TextureWrapMode.ClampToEdge);
+                    GraphicsExtensions.CheckGLError();
+                }
```

Comment:
In the non MSAA case, you are creating 2 depth/stencil buffer when only one is necessary. This block need to be merged with the depth buffer setup a few lines above.

```csharp
// pseudo code
if (depthFormat != none)
{
  // create RenderTarget.DepthStencil as a Texture2D
  depth = renderTarget.DepthStencil.glTexture;
  if (hasStencil)
  {
    stencil = renderTarget.DepthStencil.glTexture;
  }
  if (msaa)
  {
    // create renderbuffer (glRenderBufferMultisample) and override "depth" and "stencil" accordingly
  }
}

renderTarget2D.glColorBuffer = color;
renderTarget2D.glDepthBuffer = depth;
renderTarget2D.glStencilBuffer = stencil;
```